clouddriver-openstack/src/main/groovy/com/netflix/spinnaker/clouddriver/openstack/provider/agent/OpenstackServerGroupCachingAgent.groovy

```diff
@@ -0,0 +1,106 @@
+/*
+ * Copyright 2016 Target Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.spinnaker.clouddriver.openstack.provider.agent
+
+import com.netflix.frigga.Names
+import com.netflix.spinnaker.cats.agent.AgentDataType
+import com.netflix.spinnaker.cats.agent.CacheResult
+import com.netflix.spinnaker.cats.provider.ProviderCache
+import com.netflix.spinnaker.clouddriver.openstack.cache.CacheResultBuilder
+import com.netflix.spinnaker.clouddriver.openstack.cache.Keys
+import com.netflix.spinnaker.clouddriver.openstack.security.OpenstackNamedAccountCredentials
+import groovy.util.logging.Slf4j
+import org.openstack4j.model.heat.Stack
+
+import static com.netflix.spinnaker.cats.agent.AgentDataType.Authority.AUTHORITATIVE
+import static com.netflix.spinnaker.cats.agent.AgentDataType.Authority.INFORMATIVE
+
+import static com.netflix.spinnaker.clouddriver.core.provider.agent.Namespace.APPLICATIONS
+import static com.netflix.spinnaker.clouddriver.core.provider.agent.Namespace.CLUSTERS
+import static com.netflix.spinnaker.clouddriver.core.provider.agent.Namespace.SERVER_GROUPS
+
+//TODO - Implement full server group caching and on-demand caching
+@Slf4j
+class OpenstackServerGroupCachingAgent extends AbstractOpenstackCachingAgent {
+
+  static final Set<AgentDataType> types = Collections.unmodifiableSet([
+    AUTHORITATIVE.forType(SERVER_GROUPS.ns),
+    INFORMATIVE.forType(APPLICATIONS.ns),
+    INFORMATIVE.forType(CLUSTERS.ns)
```

Comment:
You should be `INFORMATIVE` for instances and load balancers as well, since you want to write instance and load balancer relationships to their respective caches here. It should be a small delta in this file (~20LOC) so don't worry about making this PR too big, and will nicely wrap up the functionality in this class.